examples/hybrid/density_current_flux_form.jl

```diff
@@ -0,0 +1,345 @@
+function energy(Yc, ρu, z)
+    ρ = Yc.ρ
+    ρθ = Yc.ρθ
+    u = ρu / ρ
+    kinetic = ρ * norm(u)^2 / 2
+    potential = z * grav * ρ
+    internal = C_v * pressure(ρθ) / R_d
+    return kinetic + potential + internal
+end
+function combine_momentum(ρuₕ, ρw)
+    Geometry.transform(Geometry.UWAxis(), ρuₕ) +
+    Geometry.transform(Geometry.UWAxis(), ρw)
+end
+function center_momentum(Y)
+    If2c = Operators.InterpolateF2C()
+    combine_momentum.(Y.ρuₕ, If2c.(Y.ρw))
+end
+function total_energy(Y)
+    ρ = Y.Yc.ρ
+    ρu = center_momentum(Y)
+    ρθ = Y.Yc.ρθ
+    z = Fields.coordinate_field(axes(ρ)).z
+    sum(energy.(Yc, ρu, z))
+end
+
+energy_0 = total_energy(Y)
+mass_0 = sum(Yc.ρ) # Computes ∫ρ∂Ω such that quadrature weighting is accounted for.
+
+function rhs!(dY, Y, _, t)
+    ρuₕ = Y.ρuₕ
+    ρw = Y.ρw
+    Yc = Y.Yc
+    dYc = dY.Yc
+    dρuₕ = dY.ρuₕ
+    dρw = dY.ρw
+    ρ = Yc.ρ
+    ρθ = Yc.ρθ
+    dρθ = dYc.ρθ
+    dρ = dYc.ρ
+
+    # spectral horizontal operators
+    hdiv = Operators.Divergence()
+    hgrad = Operators.Gradient()
+    hwdiv = Operators.WeakDivergence()
+    hwgrad = Operators.WeakGradient()
+
+    # vertical FD operators with BC's
+    vdivf2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(0.0)),
+        top = Operators.SetValue(Geometry.WVector(0.0)),
+    )
+    vvdivc2f = Operators.DivergenceC2F(
+        bottom = Operators.SetDivergence(Geometry.WVector(0.0)),
+        top = Operators.SetDivergence(Geometry.WVector(0.0)),
+    )
+    uvdivf2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(
+            Geometry.WVector(0.0) ⊗ Geometry.UVector(0.0),
+        ),
+        top = Operators.SetValue(Geometry.WVector(0.0) ⊗ Geometry.UVector(0.0)),
+    )
+    If = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    Ic = Operators.InterpolateF2C()
+    ∂ = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.WVector(0.0)),
+        top = Operators.SetValue(Geometry.WVector(0.0)),
+    )
+    ∂f = Operators.GradientC2F()
+    ∂c = Operators.GradientF2C()
+    B = Operators.SetBoundaryOperator(
+        bottom = Operators.SetValue(Geometry.WVector(0.0)),
+        top = Operators.SetValue(Geometry.WVector(0.0)),
+    )
+
+    fcc = Operators.FluxCorrectionC2C(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    fcf = Operators.FluxCorrectionF2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+
+    uₕ = @. ρuₕ / ρ
+    w = @. ρw / If(ρ)
+    wc = @. Ic(ρw) / ρ
+    p = @. pressure(ρθ)
+    θ = @. ρθ / ρ
+    Yfρ = @. If(ρ)
+
+    ### HYPERVISCOSITY
+    # 1) compute hyperviscosity coefficients
+    @. dρθ = hdiv(hgrad(θ))
+    @. dρuₕ = hdiv(hgrad(uₕ))
+    @. dρw = hdiv(hgrad(w))
+    Spaces.weighted_dss!(dYc)
+    Spaces.weighted_dss!(dρuₕ)
+    Spaces.weighted_dss!(dρw)
+
+    κ₄ = 100.0 # m^4/s
```

Comment:
Same as before, do we still need hyperdiffusion? And any reason the hyperdiffusivity is different in the flux and advection forms?

Reply:
No - the coefficients have been modified to be consistent across all 3 examples ( results in shared drive reflect this)